### PR TITLE
Migrated library from elm-export to elm-bridge for elm type generation.

### PR DESCRIPTION
**Pull request by tmbull**
I submitted issue #39 a while back, because I find `elm-bridge` to be a lot more complete in terms of the types that it supports (namely sum types). Additionally, it uses the same options for code generation as `aeson`. 

I'm not sure if this is something you are interested in or not, but I've gone ahead and done the work to migrate from `elm-export` to `elm-bridge`. 

I also have a PR open on `elm-bridge`, at which point I can update the dependency in the stack yaml.

### CODE REVIEW

File: examples/books/elm/Generated/BooksApi.elm [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+   (Json.Decode.string) >>= \pname ->
+   Json.Decode.succeed {name = pname}
 
-decodeBook : Decoder Book
-decodeBook =
-    decode Book
-        |> required "name" string
+jsonEncBook : Book -> Value
+jsonEncBook  val =
+   Json.Encode.string val.name
```
Comment (domenkozar) [inline examples/books/elm/Generated/BooksApi.elm]: Isn't this wrong, it should be an `Json.Encode.object`

**Comment by vrom911** [comment]
Any updates on this?

**Comment by mattjbray** [comment]
@tmbull thanks for putting this together. I've not tried `elm-bridge` but I'm not against supporting it in this project.

However it'd be a shame to break things for people already committed to `elm-export`. I know there is a fork out there that adds support for sum types.

I would consider a PR that factors out the type- and decoder- generation parts, leaving only the HTTP request function generation, where the user provides their own generator mechanism via `elm-export` or `elm-bridge`.

If that's too much boilerplate for the user, I'd also consider a PR that adds two new libraries, `servant-elm-export` and `servant-elm-bridge` that provide that boilerplate.

I've not looked into how technically feasible it would be to factor out the common API generation part, however.

**Comment by tmbull** [comment]
Hi @mattjbray thanks for taking a look at this. I agree that factoring out the type- and decoder- generation is the best approach.  I'll give it a try, but it will probably be a few days before I can update the PR as I'm pretty busy with other projects at the moment.

**Comment by mattjbray** [comment]
Ok, so it looks like `elm-export` is not keeping up (partly my fault as I'm a maintainer), so let's go ahead with this swap.

@tmbull can you get the travis tests to pass? It looks like `stack.yaml` is pointing to a private `elm-bridge` repo.

**Comment by k-bx** [comment]
FYI I'm continuing this work in https://github.com/haskell-servant/servant-elm/pull/49

**Pull request merged**